project_generator/tools/uvision.py

```diff
@@ -37,49 +37,128 @@ class uVisionDefinitions():
+            'uvproj': {
+                'TargetDlls': {
+                    'Driver': 'Segger\\JL2CM3.dll',
+                },
+                'Utilities': {
+                    'Flash2': 'Segger\\JL2CM3.dll',
+                },
             },
-            'Utilities': {
-                'Flash2': 'Segger\\JL2CM3.dll',
+            'uvoptx' : {
+                'DebugOpt' : {
+                    'nTsel' : '6',
+                    'pMon': 'Segger\\JL2CM3.dll',
+                },
+                'SetRegEntry' : {
+                    'Key' : 'JL2CM3',
+                },
             },
         },
         'ulink-pro': {
-            'TargetDlls': {
-                'Driver': 'BIN\\ULP2CM3.dll',
+            'uvproj': {
+                'TargetDlls': {
+                    'Driver': 'BIN\\ULP2CM3.dll',
+                },
+                'Utilities': {
+                    'Flash2': 'BIN\\ULP2CM3.dll',
+                },
             },
-            'Utilities': {
-                'Flash2': 'BIN\\ULP2CM3.dll',
+            'uvoptx' : {
+                'DebugOpt' : {
+                    'nTsel' : '7',
+                    'pMon': 'BIN\\ULP2CM3.DLL',
+                },
+                'SetRegEntry' : {
+                    'Key' : 'ULP2CM3',
+                },
             },
         },
         'st-link': {
-            'TargetDlls': {
-                'Driver': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
+            'uvproj': {
+                'TargetDlls': {
+                    'Driver': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
+                },
+                'Utilities': {
+                    'Flash2': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
+                },
             },
-            'Utilities': {
-                'Flash2': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
+            'uvoptx' : {
+                'DebugOpt' : {
+                    'nTsel' : '11',
+                    'pMon': 'STLink\\ST-LINKIII-KEIL_SWO.dll',
+                },
+                'SetRegEntry' : {
+                    'Key' : 'ST-LINKIII-KEIL_SWO',
+                },
             },
         },
         'nu-link': {
-            'TargetDlls': {
-                'Driver': 'BIN\\Nu_Link.dll',
+            'uvproj': {
+                'TargetDlls': {
+                    'Driver': 'BIN\\Nu_Link.dll',
+                },
+                'Utilities': {
+                    'Flash2': 'BIN\\Nu_Link.dll',
+                },
             },
-            'Utilities': {
-                'Flash2': 'BIN\\Nu_Link.dll',
+            'uvoptx' : {
+                'DebugOpt' : {
+                    'nTsel' : '9',
+                    'pMon': 'NULink\\Nu_Link.dll',
+                },
+                'SetRegEntry' : {
+                    'Key' : 'Nu_Link',
+                },
             },
         },
     }
 
+    # use first debugger from Keil5 debugger selection combo box as default debugger
+    debuggers_default = 'ulink2-me'
```

Comment:
wan't the default one cmsis-dap? also set in the dicts for uvproj above (=default template)?